scripts/sanity_chk/sanitylib.py

```diff
@@ -3367,6 +3366,98 @@ def csv_report(self, filename):
                     rowdict["rom_size"] = rom_size
                 cw.writerow(rowdict)
 
+    def json_report(self, filename, platform=None, append=False, version="NA"):
+        rowdict = {}
+        results_dict = {}
+        rowdict["test_suite"] = []
+        results_dict["test_details"] = []
+        new_dict = {}
+
+        if platform:
+            selected = [platform]
+        else:
+            selected = self.selected_platforms
+
+        rowdict["test_environment"] = {"os": os.name,
+                                        "zephyr_version": version,
+                                        "toolchain": self.get_toolchain()
+                                        }
+        for p in selected:
+            json_dict = {}
+            inst = self.get_platform_instances(p)
+
+            if os.path.exists(filename) and append:
+                with open(filename, 'r') as report:
+                    data = json.load(report)
+
+                for i in data["test_suite"]:
+                    test_details = i["test_details"]
+                    for test_data in test_details:
+                        if not (test_data["status"]) == "failed":
+                            new_dict = test_data
+                            results_dict["test_details"].append(new_dict)
+
+            for _, instance in inst.items():
+                handler_log = os.path.join(instance.build_dir, "handler.log")
+                build_log = os.path.join(instance.build_dir, "build.log")
+                device_log = os.path.join(instance.build_dir, "device.log")
+
+                handler_time = instance.metrics.get('handler_time', 0)
```

Comment:
Please make this double quotes to be consistent with the other `instance.metrics.get( )`